target-chains/near/receiver/src/governance.rs

```diff
@@ -312,18 +369,64 @@ impl Pyth {
     }
 
     #[private]
-    pub fn set_gov_source(&mut self, source: Source) {
-        self.gov_source = source;
+    #[handle_result]
+    pub fn authorize_gov_source_transfer(&mut self, claim_vaa: Vec<u8>) -> Result<(), Error> {
+        let (vaa, rest): (wormhole::Vaa<()>, _) =
+            serde_wormhole::from_slice_with_payload(&claim_vaa).expect("Failed to deserialize VAA");
+
+        // Convert to local VAA type to catch API changes.
+        let vaa = Vaa::from(vaa);
+
+        // Parse GovernanceInstruction from Payload.
+        let instruction =
+            GovernanceInstruction::deserialize(rest).expect("Failed to deserialize action");
+
+        // Confirm action is destined for NEAR.
+        ensure!(
+            instruction.target == Chain::from(WormholeChain::Near),
+            Unknown
+        );
+
+        // Execute the embedded VAA action.
+        match instruction.action {
+            GovernanceAction::RequestGovernanceDataSourceTransfer {
+                governance_data_source_index,
+            } => {
+                ensure!(
+                    self.executed_governance_change_vaa < governance_data_source_index as u64,
```

Comment:
i think `executed_governance_change_vaa` needs to be updated by this method (to = governance_data_source_index) after this check

Reply:
Good catch I'll add a couple of sequencing tests before merging.